x-pack/plugins/xpack_main/server/routes/api/v1/settings.js

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { wrap as wrapError } from 'boom';
+import { KIBANA_SETTINGS_TYPE } from '../../../../../monitoring/common/constants';
+import { getKibanaInfoForStats } from '../../../../../monitoring/server/kibana_monitoring/lib';
+
+const getClusterUuid = async callCluster => {
+  const { cluster_uuid: uuid } = await callCluster('info', { filterPath: 'cluster_uuid', });
+  return uuid;
+};
+
+export function settingsRoute(server, kbnServer) {
+  server.route({
+    path: '/api/settings',
+    method: 'GET',
+    async handler(req, reply) {
+      const { server } = req;
+      const { callWithRequest } = server.plugins.elasticsearch.getCluster('data');
+      const callCluster = (...args) => callWithRequest(req, ...args); // All queries from HTTP API must use authentication headers from the request
+
+      try {
+        const { collectorSet } = server.usage;
+        const settingsCollector = collectorSet.getCollectorByType(KIBANA_SETTINGS_TYPE);
+
+        const settings = await settingsCollector.fetch(callCluster);
+        const uuid = await getClusterUuid(callCluster);
+
+        const kibana = getKibanaInfoForStats(server, kbnServer);
+        reply({
+          cluster_uuid: uuid,
+          settings: {
+            ...settings,
+            kibana,
+          }
+        });
+      } catch(err) {
+        req.log(['error'], err); // FIXME doesn't seem to log anything useful if ES times out
```

Comment:
Was this `FIXME` for addressing in this PR itself or a follow up PR?

Reply:
Good question. It's a comment copied over from the other api endpoint. Let's say it will be addressed in a follow up PR

Reply:
I suspect the follow PR would need to address this problem everywhere